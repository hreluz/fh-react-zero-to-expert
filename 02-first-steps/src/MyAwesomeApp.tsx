import type { CSSProperties } from "react";

const firstName = 'Bruce';
const lastName  = 'Wayne';

const favoriteGames = ['Batman Arkham Asylum', 'Batman Arkham City', 'Batman Arkham Knight'];

const address = {
    city: 'Miami',
    state: 'California',
    zipCode: 10788
}

const isActive = true;

const myStyles:CSSProperties = {
    backgroundColor: 'red',
    borderRadius: isActive ? 10 : 20,
    padding: 10
}

export function MyAwesomeApp () {
    return (
        <>
            <div data-testid="div-app">
                <h1 data-testid="first-name-title">{firstName}</h1>
                <h3>{lastName}</h3>
                <p>{favoriteGames.join(', ')}</p>
                <p style={myStyles} >
                    {JSON.stringify(address)}
                </p>
            </div>
        </>
    )
}