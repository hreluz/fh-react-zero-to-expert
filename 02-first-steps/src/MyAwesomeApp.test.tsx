import { describe, expect, test } from "vitest";
import {render, screen} from '@testing-library/react';
import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {
        const {container} =render(<MyAwesomeApp/>)

        // screen.debug()
        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');
        
        expect(h1?.innerHTML).toContain('Bruce');
        expect(h3?.innerHTML).toContain('Wayne');
    });

    test('should render firstName and lastName - screen', () => {
        render(<MyAwesomeApp/>)

        // screen.debug()
        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Bruce');
    });

    test('should match snapshot', () => {
        const {container} = render(<MyAwesomeApp/>)
        expect(container).toMatchSnapshot();
    });

    test('should match snapshot', () => {
        render(<MyAwesomeApp/>)
        expect(screen.getByTestId('div-app')).toMatchSnapshot();
    });
});