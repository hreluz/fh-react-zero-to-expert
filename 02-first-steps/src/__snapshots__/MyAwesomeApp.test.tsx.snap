// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Bruce
    </h1>
    <h3>
      Wayne
    </h3>
    <p>
      Batman Arkham Asylum, Batman Arkham City, Batman Arkham Knight
    </p>
    <p
      style="background-color: red; border-radius: 10px; padding: 10px;"
    >
      {"city":"Miami","state":"California","zipCode":10788}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Bruce
  </h1>
  <h3>
    Wayne
  </h3>
  <p>
    Batman Arkham Asylum, Batman Arkham City, Batman Arkham Knight
  </p>
  <p
    style="background-color: red; border-radius: 10px; padding: 10px;"
  >
    {"city":"Miami","state":"California","zipCode":10788}
  </p>
</div>
`;
